### ISSUE TITLE
What are the accessibility attributes and how does it work on WatchOS 2

### ISSUE THREAD

**Issue by acegreen** [jeffargast/RSSecrets#6]
Hi,

I'm wondering if this library allows you to set the kSecAttrAccessible for the keys? I need to set certain keys after an in-app purchase on iOS then retrieve them on Apple Watch.

As stated here: https://developer.apple.com/library/watchos/documentation/General/Conceptual/AppleWatch2TransitionGuide/ManagingYourData.html

Any information on this would be greatly appreciated. Also now that WatchOS 2 is separate, do we still need to enable the keychain sharing capability?

**Comment by acegreen** [comment]
@jeffargast any feedback on this?

**Comment by jeffargast** [comment]
Hi, it doesn't allow that as it's abstracting away the underlying use of the security framework. I think you may just want to use the security APIs directly.

Jeff

**Comment by acegreen** [comment]
Hi what do you mean by abstracting away the underlying use of the security framework? There is another library called keychainAccess which allows sharing of keychain.

**Comment by jeffargast** [comment]
Let me rephrase -- as currently defined, the RSSecrets class selectors hide the fact that inside the implementation it uses SecItem calls. In that regard, it doesn't allow it. However, you could modify RSSecrets to take kSecAttr as a parameter and/or add new class selectors to achieve what you want. But at that point, it's breaking the abstraction. It sounds like you may want your client code to have more control over what happens with the security APIs, and something like KeychainAccess might be a better fit.
